fe/fe-core/src/main/java/com/starrocks/alter/Alter.java

```diff
@@ -256,54 +257,59 @@ public void processAlterMaterializedView(AlterMaterializedViewStatement stmt)
 
     private void processChangeRefreshScheme(RefreshSchemeDesc refreshSchemeDesc, MaterializedView materializedView,
                                             String dbName) throws DdlException {
-        final MaterializedView.MvRefreshScheme refreshScheme = materializedView.getRefreshScheme();
-        if (refreshSchemeDesc instanceof AsyncRefreshSchemeDesc) {
-            AsyncRefreshSchemeDesc asyncRefreshSchemeDesc = (AsyncRefreshSchemeDesc) refreshSchemeDesc;
-            final MaterializedView.AsyncRefreshContext asyncRefreshContext = refreshScheme.getAsyncRefreshContext();
-            asyncRefreshContext.setStartTime(Utils.getLongFromDateTime(asyncRefreshSchemeDesc.getStartTime()));
-            final IntLiteral step = (IntLiteral) asyncRefreshSchemeDesc.getIntervalLiteral().getValue();
-            asyncRefreshContext.setStep(step.getLongValue());
-            asyncRefreshContext.setTimeUnit(
-                    asyncRefreshSchemeDesc.getIntervalLiteral().getUnitIdentifier().getDescription());
-        }
-        MaterializedView.RefreshType oldRefreshType = refreshScheme.getType();
-        final String refreshType = refreshSchemeDesc.getType().name();
-        final MaterializedView.RefreshType newRefreshType = MaterializedView.RefreshType.valueOf(refreshType);
-        refreshScheme.setType(newRefreshType);
+        MaterializedView.RefreshType newRefreshType = refreshSchemeDesc.getType();
+        MaterializedView.RefreshType oldRefreshType = materializedView.getRefreshScheme().getType();
 
+        // TODO: The exact same refresh type does not need to drop and rebuild the task
```

Comment:
Is it easy to achieve this?

Reply:
It is not easy, because manual and asynchronous tasks can be reused, and periodically it is necessary to compare specific units to decide whether to drop and then create.